lib/githubactions/githubactions.go

```diff
@@ -0,0 +1,87 @@
+/*
+Copyright 2022 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package githubactions
+
+// GitHub Workload Identity
+//
+// GH provides workloads with two environment variables to facilitate fetching
+// a ID token for that workload.
+//
+// ACTIONS_ID_TOKEN_REQUEST_TOKEN: A token that can be redeemed against the
+// identity service for an ID token.
+// ACTIONS_ID_TOKEN_REQUEST_URL: Indicates the URL of the identity service.
+//
+// To redeem the request token for an ID token, a GET request shall be made
+// to the specified URL with the specified token provided as a Bearer token
+// using the Authorization header.
+//
+// The `audience` query parameter can be used to customize the audience claim
+// within the resulting ID token.
+//
+// Valuable reference:
+// - https://github.com/actions/toolkit/blob/main/packages/core/src/oidc-utils.ts
+// - https://docs.github.com/en/actions/deployment/security-hardening-your-deployments/configuring-openid-connect-in-cloud-providers
+
+const IssuerURL = "https://token.actions.githubusercontent.com"
+
+// IDTokenClaims is the structure of claims contained within a Github issued
+// ID token.
+//
+// See the following for the structure:
+// https://docs.github.com/en/actions/deployment/security-hardening-your-deployments/about-security-hardening-with-openid-connect#understanding-the-oidc-token
+type IDTokenClaims struct {
```

Comment:
It looks like there's a few more claims available than we opt to support/validate. Is there some reasoning to that'd be worth including in the doc comment or something?

Reply:
For now, it was mostly just a focus on adding the most "useful" fields. I don't expect anyone to say, try and filter access based on commit - but if we receive requests for this then we can support it. I'll add works to that effect.